Alter doc (#1876)
infiniflow/infinity: docs/references/pysdk_api_reference.md

---

## add_columns

```python

> table_obj.add_columns({"new_column_name1": {"type": "integer", "default": 0}, "new_column_name2": {"type": "varchar", "default": ""}})

Reviewer: ```suggestion
table_object.add_columns(columns_definition)
```

```

### Parameters

#### column_defs: `dict[str, dict[str, Any]]`, *Required*

> The same as the `column_defs` parameter in the `create_table` method. It is a dictionary where each key is a column name and each value is a dictionary containing the column's information. The default

Reviewer: ```suggestion
A dictionary defining the columns to add. Each key in the dictionary is a column name (`str`), with a corresponding 'value' dictionary defining the column's data type and default value. See the description of `create_table()`'s `columns_definition` for all available settings. 
```

> value of new columns is required.

Reviewer: ```suggestion

:::caution NOTE
You must specify a default value each time you add a column.
:::
```

> 

Reviewer: ```suggestion

#### Examples

```python
# Add an integer column and a varchar column at once
table_obj.add_columns({"new_column_name1": {"type": "integer", "default": 0}, "new_column_name2": {"type": "varchar", "default": ""}})
```
```

---

# drop_columns

```python

> table_obj.drop_columns(["column_name1", "column_name2"])

Reviewer: ```suggestion
table_object.drop_columns(columns_definition)
```

```

### Parameters

#### column_names: `list[str]`, *Required*

A list of strings representing the names of the columns to drop.

> 

Reviewer: ```suggestion

#### Examples

```python
# Remove two columns at once
table_object.drop_columns(["column_name1", "column_name2"])
```
```

---

## add_columns

```python

> table_object.add_columns(columns_definition)

Reviewer: ```suggestion
table_object.add_columns(column_defs)
```
